eth_utils/logging.py

```diff
@@ -1,19 +1,43 @@
 import contextlib
+import functools
 import logging
-from typing import Any, Iterator, Type, TypeVar
+from typing import Any, Callable, Iterator, Optional, Type, TypeVar
 
 from .toolz import assoc
 
 DEBUG2_LEVEL_NUM = 8
 
 
+class cached_show_debug2_property:
```

Comment:
Apart from typing, this looks totally generic; basically the same as `@cached_property`. Are you just trying to avoid the dependency?

If so, maybe we could just make this class into an internal `cached_property` implementation in eth-utils. Seems like we'd want to reuse it elsewhere.

Reply:
Yes, avoiding the dependency.  For libraries that really need it I have no issue adding the dependency but this is a somewhat small and specialized case and it allows us to maintain good type hints.